### DISCUSSION TITLE
How to get a list of all repeaters?

### DISCUSSION THREAD

**Discussion by nikhiltri** [area17/twill#1040] [Q&A] [answered]
We're working on migrating our codebase up from an older version of Twill. There used to be config blocks in `block_editor` that listed all blocks and repeaters, and call `config('twill.block_editor.repeaters')` or `config('twill.block_editor.blocks')` to get lists of all these resources and do things with them. These have since been replaced with annotations in block and repeater Blade files. Is there still a way to get lists of all blocks and repeaters?

**Comment by pboivin** [comment, marked as answer]
Hey @nikhiltri,

Not sure if this is exactly what you are looking for, but you can query the available repeaters and blocks from `BlockCollection` :

```
$repeaters = app(BlockCollection::class)->getRepeaters()

// and

$blocks = app(BlockCollection::class)->getBlocks()
```

Hope this can help!

**Reply by nikhiltri** [reply]
Yup, that works. I confirmed that this syntax works:

``` php
$repeatersConfig = app(BlockCollection::class)->getRepeaters();
$rep = $repeatersConfig->first(function (Block $block) {
   return $block->name == 'repeater_name___its_the_repeaters_file_name';
});
```

In place of:

``` php
$repeatersConfig = config('twill.block_editor.repeaters');
$rep = $repeatersConfig['repeater_name___key_in_block_editor_config'];
```

The old code returned an array, whereas the new code returns a Block object. So later references to `$rep` will need to be refactored.
